### ISSUE TITLE
Proposal - create wrapper around dedicated server

### ISSUE THREAD

**Issue by mithrand0** [ReactiveDrop/reactivedrop_public_src#627]
When running a server, there are a lot things that need to be taken care of, like grabbing custom scripts from the internet for reboot loops. Also, we have the ongoing issue of workshop items not working sometimes.

This can be all solved by creating a wrapper ourselves. We could make a small `rd_server.exe` that does the following:

```
> start
  > update wachter
     > [if update / or daily at predefined time]
       > steamcmd update game
       > steamcmd update workshop items
  
  > [watchdog]
     > start server 1
  > [watchdog]
     > start server 2
```

*update wacher*

Can watch the Steam manifest data every x minutes for updates, or schedule a full check (with validate) at a predefined time. When an update is received, it waits until all servers are down/empty, then updates before restarting them.

*steamcmd*

Can download both game and workshop updates:

> steamcmd ... app_update 563560
> steamcmd ... app_update 1007 
> steamcmd ... workshop_download_item 563560 <workshop-id>
> steamcmd ... workshop_download_item 563560 <workshop-id>

One thing that would need to change with this approach, is that the game doesn't try to download/install workshop items by itself in server mode. It simply needs to download a pre-existing file on the filesystem. The directory format of 'workshop_download_item' seems to be example the same as what the server would do, it's just another folder.

*server watchdog*

Monitors the server by connecting to it every minute. If the server is down or frozen, it restarts them. If an update is pending, it waits restarting them until the update is done.

*config*

For the config, we can use the same mechanics Left for Dead has implemented with the `-fork` parameter, which spawns multiple servers. Options given to `rd_server.exe` are passed to `srcds_console.exe`. Ports are handled the same way `-fork` would do.

*TLDR*

Creating this can make it easier for people who want to run a dedicated server, who don't have the knowledge to do so. They don't need to care about downloading stuff like steamcmd, installing a server, keeping it updated and maintained. We can also add sane default values to it before it runs the first time.

**Comment by mithrand0** [comment]
Implemented:

https://github.com/mithrand0/srcds-guardian
